examples/enc-search-sample.rs

```diff
@@ -0,0 +1,421 @@
+            .read_line(&mut name)
+            .expect("error: couldn't read input");
+        name = name.trim().to_string();
+        if &name == "" {
+            break;
+        } else {
+            let mut email = String::new();
+            println!("Enter customer's email");
+            std::io::stdin()
+                .read_line(&mut email)
+                .expect("error: couldn't read input");
+            email = email.trim().to_string();
+
+            let cust = encrypt_customer(&sdk, next_id, &name, &email, &group_id).await?;
+            save_customer(cust, &[], &[]);
+        }
+        next_id += 1;
+    }
+
+    // now create a new customer record to index
+    let mut customer = Customer {
+        id: 23,
+        name: "J. Fred Muggs".to_string(),
+        name_keys: "".to_string(),
+        email: "j.fred.muggs@nowhere.com".to_string(),
+        email_keys: "".to_string(),
+    };
+
+    // Generate the index tokens for the customer name, then encrypt it
+    // start-snippet{indexData}
+    let name_tokens = blind_index
+        .tokenize_data(&customer.name, None)?
+        .into_iter()
+        .collect::<Vec<u32>>();
+    let group_id = create_group_id("customerService")?;
+    let encrypt_opts = DocumentEncryptOpts::with_explicit_grants(
+        None,                  // document ID - create unique
+        None,                  // document name
+        false,                 // don't encrypt to self
+        vec![group_id.into()], // users and groups to which to grant access
+    );
+    let enc_name = sdk
+        .document_encrypt_unmanaged(customer.name.as_bytes(), &encrypt_opts)
+        .await?;
+    // Replace name with encoded encrypted version. Also need to store EDEKs to decrypt name.
+    customer.name = base64::encode(enc_name.encrypted_data());
+    customer.name_keys = base64::encode(enc_name.encrypted_deks());
+    save_customer(customer.clone(), &name_tokens, &[]);
+    // end-snippet{indexData}
+
+    // Now ask the user for a query string, ask the server for matching records,
+    // and display all matches to the user
+    // start-snippet{executeSearch}
+    let query_str = get_search_query();
+    display_matching_customers(&sdk, &blind_index, &query_str).await?;
+    // end-snippet{executeSearch}
+
+    // Now create a second blind index to use for the email address fields.
+    // start-snippet{createSecondIndex}
+    let encrypted_salt2 = sdk.create_blind_index(&salt_group_id).await?;
+    let blind_index2 = encrypted_salt2.initialize_search(&sdk).await?;
+    // end-snippet{createSecondIndex}
+
+    let mut customer2 = Customer {
+        id: 24,
+        name: "Pokey".to_string(),
+        name_keys: "".to_string(),
+        email: "pokey@gumby.io".to_string(),
+        email_keys: "".to_string(),
+    };
+
+    // start-snippet{useSecondIndex}
+    // Generate the index tokens for the customer name and email address, then encrypt them
+    let name_tokens = blind_index
```

Comment:
This is the same thing you just did above, it'd be better if we refactored it out to take a `Customer` and returned a `EncryptedCustomer` with the 2 sets of index tokens.